src/domain/dataverse/domain.ts

```diff
@@ -90,15 +96,29 @@ export const storeFactory = (
             limit: 20,
             hasNext: false,
             error: O.none,
-            filters: { byType: 'all' },
+            filters: { byType: 'all', byProperty: null },
             language: 'en'
           }))
         ),
         dataverse: () => () => get().data.dataverse,
         hasNext: () => () => get().data.hasNext,
         isLoading: () => () => get().data.isLoading,
         error: () => () => get().data.error,
-        filters: () => () => get().data.filters,
+        byTypeFilter: () => () => get().data.filters.byType,
+        byPropertyFilter: () =>
+          pipe(
+            () => get().data.filters.byProperty,
+            IO.flatMap(x =>
+              pipe(
+                x,
+                O.fromNullable,
+                O.match(
+                  () => IO.of(''),
+                  filter => IO.of(filter.value)
+                )
+              )
+            )
+          ),
```

Comment:
A little too complicated for my liking. Let's try to simplify using the `flow` operator instead of `pipe`, and by avoiding the unecessary `flatMap`, where a simple `map` is sufficient:

```suggestion
        byPropertyFilter: () =>
          pipe(
            () => get().data.filters.byProperty,
            IO.map(
              flow(
                O.fromNullable,
                O.match(
                  () => '',
                  filter => filter.value
                )
              )
            )
          ),
```